pybamm/expression_tree/concatenations.py

```diff
@@ -338,7 +338,7 @@ def create_slices(self, node: pybamm.Symbol) -> defaultdict:
                 """Concatenation and children must have the same number of
                 points in secondary dimensions"""
             )
-        for i in range(second_pts):
+        for _i in range(second_pts):
```

Comment:
Is this setting optional? Do all loop variables need this now?

Reply:
This should be the rule: https://docs.astral.sh/ruff/rules/unused-loop-control-variable/. I don't think all loop variables need it and it may not be required in things like notebooks, but for a lengthy file with multiple classes and methods I think it should be fine

Reply:
I think if it is just for unused variables then it should be "_" instead of "_i", which is pretty common. Using "_i" just seems weird if there is only a single variable. Not sure how configurable it is though.

When I saw it I was concerned that it was done for scope issues, which are better resolved by shortening functions.

Reply:
You're right. `_i` seems to be more like a private variable – not something that is meant to be discarded after an iterable has been called over, `_` is better for the latter

Reply:
I have corrected the naming convention for this in f6b7b8cc11b6290737e694ac92ca4a8e93ba4138